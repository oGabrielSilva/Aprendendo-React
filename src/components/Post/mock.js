export const mockPost = {
	posts: [
	{
		id: 1,
		title: 'Title 1',
		body: 'Body 1',
		cover: 'static/logo1.svg'
	},
	{
		id: 2,
		title: 'Title 2',
		body: 'Body 2',
		cover: 'static/logo2.svg'
	},
	{
		id: 3,
		title: 'Title 3',
		body: 'Body 3',
		cover: 'static/logo3.svg'
	}
	]
};
