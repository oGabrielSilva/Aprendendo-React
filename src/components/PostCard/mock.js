export const mockPostCard = {
	post: {
		title: 'Title 1',
		body: 'Body 1',
		id: 1,
		cover: 'static/logo.svg'
	}
};
