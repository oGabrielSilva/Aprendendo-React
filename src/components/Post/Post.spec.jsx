import { render, screen } from '@testing-library/react';
import { Posts } from '.';
import { mockPost as mock } from './mock';

const props = mock;

describe('<Post />', () => {
	it('should render Post', () => {
		render(<Posts {...props} />);
		expect(screen.getAllByRole('heading', { name: /title/i }))
			.toHaveLength(3);

		expect(screen.getAllByRole('img', { name: /title/i }))
			.toHaveLength(3);

		expect(screen.getAllByText(/body/i))
			.toHaveLength(3);

		expect(screen.getByRole('img', { name: /title 3/i }))
			.toHaveAttribute('src', 'static/logo3.svg');
	});

	it('should match snapshot', () => {
		const { container } = render(<Posts  {...props} />);
		expect(container.firstChild).toMatchSnapshot();
	})
});
