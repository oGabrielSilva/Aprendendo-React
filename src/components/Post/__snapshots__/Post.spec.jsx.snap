// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Title 1"
      src="static/logo1.svg"
    />
    <div
      class="post-content"
    >
      <h1>
        1 Title 1
      </h1>
      <p>
        <span>
          Body 1
        </span>
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 2"
      src="static/logo2.svg"
    />
    <div
      class="post-content"
    >
      <h1>
        2 Title 2
      </h1>
      <p>
        <span>
          Body 2
        </span>
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 3"
      src="static/logo3.svg"
    />
    <div
      class="post-content"
    >
      <h1>
        3 Title 3
      </h1>
      <p>
        <span>
          Body 3
        </span>
      </p>
    </div>
  </div>
</div>
`;
